src/du/du.rs

```diff
@@ -91,16 +91,23 @@ fn du(mut my_stat: Stat, options: &Options, depth: usize) -> Box<DoubleEndedIter
         };
 
         for f in read.into_iter() {
-            let entry = crash_if_err!(1, f);
-            let this_stat = Stat::new(entry.path());
-            if this_stat.is_dir {
-                futures.push(du(this_stat, options, depth + 1));
-            } else {
-                my_stat.size += this_stat.size;
-                my_stat.blocks += this_stat.blocks;
-                if options.all {
-                    stats.push(this_stat);
-                }
+            match f {
+                Ok(entry) => match fs::symlink_metadata(entry.path()) {
```

Comment:
I’d prefer if you just change `Stat::new()` so that it returns `io::Result<Stat>`.  That way we won’t have to call `fs::symlink_metadata()` twice for every entry.

Reply:
I considered that but thought it would be weird for a constructor to return a result - Is it considered generally ok to do this in rust?

No problem though I'm happy to change it.

Reply:
It’s weirder for it to abort the program IMO.  I think it’s fine, but you could change `new` to `from_entry` or something if you’d prefer.